src/Middleware/RetryMiddleware.php

```diff
@@ -155,4 +167,26 @@ protected function getCurrentTimeMs()
     {
         return microtime(true) * 1000.0;
     }
+
+    /**
+     * This is the default retry behaviour.
+     */
+    private function getRetryFunction()
+    {
+        return $this->retrySettings->getRetryFunction() ??
+            function (\Exception $e, array $options, int $retryAttempt): bool {
```

Comment:
We are passing in `$retryAttempts` but not using it? I am assuming this is because we are going to pass in custom retry functions in handwritten libraries which support the `$retries` argument? But if we do that, we might as well just add a `maxRetries` option to `RetrySettings` and support it as well.  What's the intention here?

Reply:
Update on this. We finally did add the maxRetries as we saw many uses for that. This is however considered experimental for now.